utils/patch-for-qa.sh

```diff
@@ -0,0 +1,85 @@
+    # Adding accessPortal, configServer, notificationServer, securityRuleSsh to variables section.
+    jq ".variables.hostSetup.accessPortal = \"${accessPortal}\"" ${MAIN_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${MAIN_TEMPLATE}
+    jq ".variables.hostSetup.configServer = \"${configServer}\"" ${MAIN_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${MAIN_TEMPLATE}
+    jq ".variables.hostSetup.notificationServer = \"${notificationServer}\"" ${MAIN_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${MAIN_TEMPLATE}
+    jq ".variables.securityRuleSsh = ${SR_SSH}" ${MAIN_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${MAIN_TEMPLATE}
+
+    # Adding customData, securityGroupSsh to parameters section.
+    jq ".parameters.customData = ${CUSTOM_DATA_PARAMETER}" ${MAIN_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${MAIN_TEMPLATE}
+    jq ".parameters.securityGroupSsh = ${SG_SSH}" ${MAIN_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${MAIN_TEMPLATE}
+
+    # Updating customDataWithoutHttpProxy, securityRules variables.
+    jq ".variables.customDataWithoutHttpProxy = \"${CUSTOM_DATA_WO_HTTP_PROXY}\"" ${MAIN_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${MAIN_TEMPLATE}
+    jq ".variables.securityRules = \"${SR}\"" ${MAIN_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${MAIN_TEMPLATE}
+
+    # Updating customData for VM resource.
+    jq "( .resources[] | select(.name == \"SettingUpVirtualMachine\") ).properties.parameters.osProfile.value.customData |= \"${CUSTOM_DATA_VM}\"" ${MAIN_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${MAIN_TEMPLATE}
+}
+
+function patchUITemplate() {
+    # Adding ssh rule option to securityGroup selectbox.
+    jq "( .parameters.steps[0].elements[] | select(.name == \"securityGroup\") ).constraints.allowedValues += ${UI_SG_SSH_OPTION}" ${UI_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${UI_TEMPLATE}
+    jq "( .parameters.steps[0].elements[] | select(.name == \"securityGroup\") ).multiselect = \"true\"" ${UI_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${UI_TEMPLATE}
+
+    # Adding injecting radio input.
+    jq ".parameters.basics += ${UI_INJECTING_FIELD}" ${UI_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${UI_TEMPLATE}
+
+    # Adding Custom data field.
+    jq ".parameters.basics += ${UI_CUSTOM_DATA_FIELD}" ${UI_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${UI_TEMPLATE}
+
+    # Updating visible option for jointoken field.
+    jq "( .parameters.basics[] | select(.name == \"jointoken\") ).visible |= \"${UI_JOINTOKEN_VISIBLE}\"" ${UI_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${UI_TEMPLATE}
+
+    # Adding scustomData, ecurityGroupSsh to output section.
+    jq ".parameters.outputs.customData = \"${UI_CUSTOM_DATA_OUTPUT}\"" ${UI_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${UI_TEMPLATE}
+    jq ".parameters.outputs.securityGroupSsh = \"${UI_SG_SSH_OUTPUT}\"" ${UI_TEMPLATE} > tmp.$$.json && mv tmp.$$.json ${UI_TEMPLATE}
+}
+
+function usage() {
+    echo -e "\nUsage: $0 accessPortal=<accessPortal> configServer=<configServer> notificationServer=<notificationServer>"
```

Comment:
by passing `accessPortal` info here itself this archive will be bind to 1 environment right, why not create a template without populating  `accessportal` info and QA uses cli or UI to update the same.

Reply:
Yes, we can add these fields as parameters for Main template and add them to UI template too. I'll update it.

Reply:
Are you updating this change? I still see the script need `accessportal` info to run this script